### ISSUE TITLE
ArConfig_setUpdateMode has no effect on ArSession_update execution time

### ISSUE THREAD

**Issue by blackarrow13** [google-ar/arcore-android-sdk#1012]
### SPECIFIC ISSUE ENCOUNTERED
I have measured the time it takes for `ArSession_update(ar_session_, ar_frame_)` call to finish - it seems unreasonably slow, ~15-40 ms. Using `ArConfig_setUpdateMode(ar_session_, out_config, AR_UPDATE_MODE_LATEST_CAMERA_IMAGE);` has no effect, despite what documentation says.
More details in "Steps to reproduce".

### VERSIONS USED
- Android Studio: 3.6
- ARCore SDK for Android: 1.15.0
- Device manufacturer, model, and O/S: multiple devices.
Google Pixel 2XL, Samsung Galaxy s10e, Samsung Galaxy Tab s5e, Huawei Nexus 6P, Asus_A002, Sony Xperia XZ3
- ARCore: 1.15200121046 on all devices

### STEPS TO REPRODUCE THE ISSUE
 1. Open the sample project in Android Studio https://github.com/google-ar/arcore-android-sdk/tree/master/samples/hello_ar_c
 2. Add a log before and after ArSession_update call (https://github.com/google-ar/arcore-android-sdk/blob/master/samples/hello_ar_c/app/src/main/cpp/hello_ar_application.cc#L131)
 3. Run the app on a device and look at the logs timestamp difference.
In my case, on Google Pixel 2XL it takes about 1-3ms, on every other device - about 15-40ms, sometime more.

Now, by default this call is waiting for a new frame to arrive, so that may be ok.
I have also added the following to OnSurfaceCreated() method:

```
    ArConfig* out_config;
    ArConfig_create(ar_session_, &out_config);
    ArConfig_setUpdateMode(ar_session_, out_config, AR_UPDATE_MODE_LATEST_CAMERA_IMAGE);
```
According to the documentation, ArSession_update should "return immediately without blocking". - this is not the case on most of my devices.
On Google Pixel 2XL, the call still takes 1-3ms (fine).
On Samsung Galaxy Tab s5e, s10e, on Asus_A202 and on Huawei 6P the call still takes 15-40ms.
Sony Xperia XZ3 is the only device where this setting seem to have an effect - the execution time of ArSession_update drops from ~30ms to ~2ms.

In short, 2 things seems wrong: 1) on most devices update call takes very long (why is Pixel so different?) 2) the respective config setting does nothing (except one device)

**Comment by devbridie** [comment]
Thanks for the detailed report. I will procure a device and try to reproduce ASAP.

**Comment by devbridie** [comment]
Indeed, [AR_UPDATE_MODE_LATEST_CAMERA_IMAGE](https://developers.google.com/ar/reference/c/group/config#arupdatemode) seems to have no effect on some of these devices. I'll update when I have more information for you.

**Comment by blackarrow13** [comment]
Thank you for the quick reaction!
In the meantime I've also tested this on Samsung Galaxy S9 - same behavior, both with Android 9 and Android 10 OS.
Looking forward to updates.

**Comment by devbridie** [comment]
Just to be certain: are you using `ArSession_configure` to apply your modification to the configuration to your current session? That is, the added code should be the following:

```
    ArConfig* out_config;
    ArConfig_create(ar_session_, &out_config);
    ArConfig_setUpdateMode(ar_session_, out_config, AR_UPDATE_MODE_LATEST_CAMERA_IMAGE);
    ArSession_configure(ar_session_, out_config); // apply configuration to session
```

**Comment by blackarrow13** [comment]
I wasn't calling ArSession_configure. I wrongly assumed it already sets the mode since ArConfig_setUpdateMode already takes an ar_session.
My mistake, sorry for that - and thank you for taking a look at this.
After calling ArSession_configure the update() call takes around 1ms on the devices, as expected.

**Comment by devbridie** [comment]
No problem! I think our documentation isn't too clear on this. We'll update the documentation to be more specific in that `ArSession_configure` should be used to apply changes located in a config.
